torch/csrc/autograd/custom_function.h

```diff
@@ -11,4 +14,276 @@ TORCH_API variable_list _wrap_outputs(
   const std::unordered_set<at::TensorImpl*> &dirty_inputs,
   const at::ArrayRef<Variable> raw_outputs,
   const std::shared_ptr<Node> &cdata);
+
+// To use custom autograd operations implement a CFunction subclass with
+// static backward and forward functions
+//
+// forward() can take as many arguments as you want and should return a
+// variable list. Use of any direct Variable arguments will be registered in
+// the graph but no vectors/sets or any other data structures will be traversed.
+// It should take an AutogradContext* as the first argument. Variables can be
+// saved in the ctx using save_for_backward() and other data can be saved in the
+// map ctx.save in the form of <std::string, at::IValue> pairs.
+//
+// backward() should take an AutogradContext* and a variable list containing as
+// many Variables as there were outputs from forward as arguments. It should
+//  return as many Variables as there were inputs with each of them containing
+// the gradient w.r.t. its corresponding input. Variables saved in forward can
+// be accessed with ctx->get_saved_variables() and other saved data can be
+// accessed from ctx->save.
+//
+// For example:
+// class MyFunction : public Function<MyFunction> {
+//   public:
+//   static variable_list forward(AutogradContext *ctx, int n, Variable var) {
+//      // Save data for backward in context
+//      ctx->save["n"] = n;
+//      var.mul_(2);
+//      // Mark var as modified by inplace operation
+//      ctx->mark_dirty({var});
+//      return std::vector<Variable>({var});
```

Comment:
nit: std::vector<Variable> -> variable_list ?
And same for the backward if it is correct.